vnext/Microsoft.ReactNative/Fabric/Composition/CompositionViewComponentView.cpp

```diff
@@ -1268,6 +1271,30 @@ void ViewComponentView::updateLayoutMetrics(
   Visual().Size(
       {layoutMetrics.frame.size.width * layoutMetrics.pointScaleFactor,
        layoutMetrics.frame.size.height * layoutMetrics.pointScaleFactor});
+
+  if (m_shouldClipChildren) {
+    // Convert layout size from logical units to physical pixels, accounting for DPI
+    const float width = layoutMetrics.frame.size.width * layoutMetrics.pointScaleFactor;
+    const float height = layoutMetrics.frame.size.height * layoutMetrics.pointScaleFactor;
+
+    // Resolve border metrics including corner radii
+    const auto borderMetrics = BorderPrimitive::resolveAndAlignBorderMetrics(layoutMetrics, *m_props);
+
+    // Create a clipping path as a rounded rectangle (or regular rectangle if radii are zero)
+    const winrt::com_ptr<ID2D1PathGeometry> pathGeometry = BorderPrimitive::GenerateRoundedRectPathGeometry(
+        m_compContext,
+        borderMetrics.borderRadii, // Controls corner roundness
+        {0, 0, 0, 0}, // Border thickness (not relevant for clipping)
+        {0, 0, width, height} // Shape size for the clipping area
+    );
+
+    // Apply the clipping path to the visual
+    Visual().as<::Microsoft::ReactNative::Composition::Experimental::IVisualInterop>()->SetClippingPath(
```

Comment:
We already setting the clipping path when we have borderRadii, in ComponentView::updateClippingPath.  -- I think that is done today to ensure that the background has a rounded corner too.

So I think the fix here will be more complex than this.